### ISSUE TITLE
FrameBuffer is not fully deallocated - Cocos2d-x v3.8 

### ISSUE THREAD

**Issue by rauluranga** [cocos2d/cocos2d-x#13816]
Hi Guys!
I just found that the `_defaultFBO` property (a FrameBuffer) on the `CCDirector` class is not fully deallocated. I was trying to integrate UIKit with the latest Cocos2d-x v3.8 and everything was fine but re-assign the CCEAGLView.
This bug is because the `FrameBuffer` has two static variables:

``` cpp
FrameBuffer* FrameBuffer::_defaultFBO = nullptr;
std::set<FrameBuffer*> FrameBuffer::_frameBuffers;
```

and they need to be cleared when you re-assign the CCEAGLView.   

so i added two things:

First on **`CCFrameBuffer`** class [added the following method](https://github.com/rauluranga/Cocos2dx-UIKit/blob/master/cocos2d/cocos/renderer/CCFrameBuffer.cpp#L295):

``` cpp
// .
// .
// .
void FrameBuffer::clean() {
    if(_defaultFBO)
    {
        _defaultFBO = nullptr;
    }
    _frameBuffers.clear();
}
```

and the corresponding [header declaration](https://github.com/rauluranga/Cocos2dx-UIKit/blob/master/cocos2d/cocos/renderer/CCFrameBuffer.h#L199)

``` cpp
// .
// .
// .
public:
    static FrameBuffer* getOrCreateDefaultFBO(GLView* glView);
    static void clean();
```

Last on **`CCDirector`** class deconstructor [make the call](https://github.com/rauluranga/Cocos2dx-UIKit/blob/master/cocos2d/cocos/base/CCDirector.cpp#L188-190) and release `_defaultFBO` property instead of delete:

``` cpp
   //CC_SAFE_DELETE(_defaultFBO);
    CC_SAFE_RELEASE(_defaultFBO);
    experimental::FrameBuffer::clean();
```

i don't know if this is the best way to solve it, but it just works!

final [working example](https://github.com/rauluranga/Cocos2dx-UIKit).

**Comment by winneryong** [comment]
today, i find this bug too. thank your solution

**Comment by pandamicro** [comment]
@ricardoquesada @super626 Can you confirm this issue ?

**Comment by ricardoquesada** [comment]
@pandamicro  confirmed. it is a bug.

**Comment by Odinysus** [comment]
thanks for your report! it help me much!

**Comment by AlwynC** [comment]
@ricardoquesada @pandamicro @super626 This issue has been fixed in pull request #14471.I've tested for this issue in latest version cocos2d-x v3.11.1 and it does not seem to be appearing there.Why is this issue still open then?

**Comment by ricardoquesada** [comment]
let's close then. thanks.
